src/utils/dm/findDMForUser.ts

```diff
@@ -71,7 +84,10 @@ function extractSuitableRoom(rooms: Room[], userId: string): Room | undefined {
 export function findDMForUser(client: MatrixClient, userId: string): Room | undefined {
     const roomIdsForUserId = DMRoomMap.shared().getDMRoomsForUserId(userId);
     const roomsForUserId = roomIdsForUserId.map((id) => client.getRoom(id)).filter((r): r is Room => r !== null);
-    const suitableRoomForUserId = extractSuitableRoom(roomsForUserId, userId);
+    // Call with findRoomWithThirdpartyInvites = true to also include rooms with pending thirdparty invites.
+    // roomsForUserId can only contain rooms with the other user here,
+    // because they have been queried by getDMRoomsForUserId().
+    const suitableRoomForUserId = extractSuitableRoom(roomsForUserId, userId, true);
 
     if (suitableRoomForUserId) {
         return suitableRoomForUserId;
```

Comment:
I'd suggest, to make it more clear what behavior has changed, to change the fallback in all rooms to

```ts
return extractSuitableRoom(allRooms, userId, false);
```